cxx/non-atomic-action-interpreter-module/utils/TemplateParamsUtils.cpp

```diff
@@ -0,0 +1,36 @@
+#include "TemplateParamsUtils.hpp"
+
+using namespace nonAtomicActionInterpreterModule;
+
+ScTemplateParams TemplateParamsUtils::createTemplateParamsFromReplacements(
+    std::map<ScAddr, ScAddr, ScAddrLessFunc> const & replacements,
+    std::map<ScAddr, std::string, ScAddrLessFunc> const & resolvedVariableIdentifiers)
+{
+  ScTemplateParams templateParams;
+  for (auto const & variableReplacementPair : replacements)
+  {
+    std::string const & variableIdentifier = resolvedVariableIdentifiers.at(variableReplacementPair.first);
+    templateParams.Add(variableIdentifier, variableReplacementPair.second);
```

Comment:
```suggestion
    templateParams.Add(variableReplacementPair.first, variableReplacementPair.second);
```
now sc-machine allows to add template param by var
maybe system idtf map can be deleted, can you try and get rid of it?